### PR TITLE
Make sure we only show the modal when the data was actually fetched.

### PR DESCRIPTION
**Pull request by javierarce**
cc @xavijam

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4219/))

File: lib/assets/javascripts/cartodb/table/mapview.js [hunk 1]
```diff
@@ -650,7 +653,7 @@ cdb.admin.MapTab = cdb.core.View.extend({
   },
 
   _showScratchDialog: function() {
-    if (this.user.featureEnabled('new_modals') && this.options.table && this.options.table.data().length === 0) {
+    if (this.user.featureEnabled('new_modals') && this.options.table && this.options.table.data().fetched && this.options.table.data().length === 0) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/mapview.js]: ```this.table``` should be defined here.

**Comment by xavijam** [comment]
I've checked in that scratch dialog [code](https://github.com/CartoDB/cartodb/blob/3985-fix-scratch-modal-popup/lib/assets/javascripts/cartodb/table/mapview.js#L669), there is a ```var self = this```, i think we can remove it just adding  
this in the bind scope as:
```js
this.view.bind('whatever', function() { ... }, this);
```

File: lib/assets/javascripts/cartodb/table/mapview.js (hunk 1 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb/table/mapview.js]: No, it's defined inside of a binding here: https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/table/mapview.js#L1401 so it's not available always (in my code, it's when calling the ```_showScratchDialog``` from the ```bindGeoRefCheck``` method, but not from ```setupMap```)

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/mapview.js]: oook, got it :+1:

**Comment by javierarce** [comment]
Ok, I'll remove the ```self``` :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4220/))

**Pull request merged**